docs/traffic-mirroring-proposal.md

```diff
@@ -0,0 +1,1282 @@
+3. In the test-cases below we are considering `ovs-mirror` plugin as the only responsible plugin for the mirror management for both producers and consumers but, as detailed in the “Pros and Cons” paragraph at the end of the proposal, it is also reasonable to consider the implementation of two distinct plugins (`ovs-mirror-producer` and `ovs-mirror-consumer`).
+4. In all diagrams below we used different colors to represent the logical relation between different entities. In case of OVS they are real DB relations, in case of Pods they represent network connections. Instead, NADs are represented with random colors without a real meaning.
+5. In all diagrams below we focused on OVS Mirror `src_port` and `dst_port` to consider the representation with the finest granularity. In this way, we can specify single ports one by one. 
+However, we are not expert in OVS and its database. In the official [DB Schema 7.10.1 PDF of Open vSwitch 2.3.90](http://www.openvswitch.org//ovs-vswitchd.conf.db.5.pdf) there is a detailed documentation about database relations and columns. Page 41 explains the *Mirror* table, but the *Selecting Packets for Mirroring* paragraph seems a little bit confusing. It’s not clear if `select_*` columns are mutually exclusive (like for outputs) or if there is any sort of priority. What happens if we set `select_all=true` with also a port in `select_src_port` and `select_dst_port`? The same applies for `select_vlan`.
+Also, we have no experience in RSPAN ports (used in conjunction with `output_vlan`). In fact, we considered only `output_port` in all our examples.
+6. We chose `selection` and `output` as JSON keys to refer to input and output ports for the mirror, and `mirrors` key to define the mirrors. However, these are only example keywords of this proposal. We should define better names with you and the rest of the community. For example, we can use specific keys like `producers` and `consumer`, instead of `mirrors`.
+
+### **1. NAD approach**
+
+In this approach the complexity is shifted toward the NAD configuration side.  The more the scenario is complex (e.g. a lot of vlans, multiple mirrors and different behaviour between pods) the more a greater number of NADs must be defined prior to the pods deployment (could become a little bit messy in complex cases).
+
+**Producer NAD**
+
+```json
+{
+    "type": "ovs-mirror",
+    "bridge": BRIDGE_NAME,
+    "mirrors": [
+        {
+            "name": MIRROR_NAME,
+            "selection": SELECTION_ARRAY
+        },
+        (...)
+    ]
+}
+```
+
+`BRIDGE_NAME`: string that represents the unique name of the bridge in ovs database where the mirror should be added
+
+`MIRROR_NAME`: string that represents the unique name of the mirror in ovs database
+
+`SELECTION_ARRAY`: array of strings that represents the type of selections. Allowed values are:
+- `src_port`
+- `dst_port`
+- `all`
+- `vlan`
```

Comment:
Probably we should only consider dst_port and src_port, since with these two options you can build the most fine grained configuration for the mirrors. The only important aspect is to allow the users to set both src and dst at the same time.

Reply:
I see, thanks